.github/workflows/push-pr-lint.yaml

```diff
@@ -0,0 +1,57 @@
+name: lint, test and build image
+on: [pull_request, push]
+
+jobs:
+  lint-test:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Install Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: '1.20'
+    - name: Checkout code
+      uses: actions/checkout@v3
+    - name: golangci-lint
+      uses: golangci/golangci-lint-action@v3
+      with:
+        args: --config .golangci.yml --timeout 2m
+        version: v1.52.2
+    - name: Test
+      run: go test ./...
+  build:
+    runs-on: ubuntu-latest
+    needs: [lint-test]
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.20'
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: build binary
+        run: make build-linux
+
+      - name: Set up Docker Buildx
```

Comment:
all the docker/image bits can be excluded

Reply:
kept these is as per our discussion on slack